tap_file/tap.py

```diff
@@ -14,31 +13,52 @@ class TapFile(Tap):
 
     name = "tap-file"
 
-    # TODO: Update this section with the actual config values you expect:
     config_jsonschema = th.PropertiesList(
         th.Property(
-            "auth_token",
+            "protocol",
             th.StringType,
             required=True,
-            secret=True,  # Flag config as protected.
-            description="The token to authenticate against the API service",
+            allowed_values=["file", "s3"],
+            description="The protocol to use to retrieve data. One of `file` or `s3`.",
         ),
         th.Property(
-            "project_ids",
-            th.ArrayType(th.StringType),
+            "filepath",
+            th.StringType,
             required=True,
-            description="Project IDs to replicate",
+            description="The path to obtain files from. Example: `/foo/bar`.",
+        ),
+        th.Property(
+            "file_regex",
+            th.RegexType,
+            description=(
+                "A regex pattern to only include certain files. Example: `.*\\.csv`."
+            ),
         ),
         th.Property(
-            "start_date",
-            th.DateTimeType,
-            description="The earliest record date to sync",
+            "s3_anonymous_connection",
+            th.BooleanType,
+            default=False,
+            description=(
+                "Whether to use an anonymous S3 connection, without the use of any "
+                "credentials. Ignored if `protocol!=s3`."
+            ),
+        ),
+        th.Property(
+            "s3_access_key",
+            th.StringType,
+            description=(
+                "The access key to use when authenticating to S3. Ignored if "
+                "`protocol!=s3` or `s3_anonymous_connection=True`."
+            ),
         ),
         th.Property(
-            "api_url",
+            "s3_access_key_secret",
```

Comment:
I think that keeping separate config options for protocol, filepath, file_regex makes sense, instead of putting it all into a URI. That's a few extra config options, but:

- It's a one-time cost. No extra config options will be required as the number of supported sources increases.
- It's easier to implement and maintain on the backend, as each config option can be referenced individually rather than having to parse a URI,
- (opinion-based) It's a better experience for the user and is less likely to be the source of error.

As for `s3_anonymous_connection`, `s3_access_key`, and `s3_access_key_secret`, they could all be ditched if users are told to use the appropriate environment variables. I think that's probably the better long-term solution. It also has the additional advantage of easily allowing other S3 authentication methods supported by boto, such as security tokens, with no extra development work in the tap itself.